### PR TITLE
PRO-1294 Issues - Pre-select best offer for swaps to estimate transaction

### PR DESCRIPTION
**Pull request by lalitkumarrock**
https://linear.app/pillarproject/issue/PRO-1294/pillar-wallet-pre-select-best-offer-for-swaps-to-estimate-transaction

## Description

- Transaction fee manage with redux
- When tap on offer then calculate that offer transaction fee
- Press "See other offers" to not recalculate transaction fee

## Motivation and Context

-  https://linear.app/pillarproject/issue/PRO-1294#comment-38d07be6

## How Has This Been Tested?

- Tested on Simulator 14 Pro
https://bscscan.com/tx/0xc4f82b9d74a8fc6b9ee0170659925001cbced35bfe2bbba595a97c6ebe160837

## Screenshots (if appropriate):

https://github.com/pillarwallet/pillarwallet/assets/82652040/e9f8dd5b-e605-46ec-8335-f14f40e17c50

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/reducers/exchangeGasFeeReducer.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  action: ExchangeGasFeeReducerAction,
+) {
+  const { data } = state;
+  switch (action.type) {
+    case ADD_EXCHANGE_GAS_FEE_INFO:
+      const { data: addFeeInfo } = action.payload;
+
+      const { provider, chain, gasFeeAsset } = addFeeInfo;
+
+      const index = state.data?.findIndex(
```
Comment (IAmKio) [inline src/reducers/exchangeGasFeeReducer.tsx:54]: Have we handle what happens if `state.data` is `null` and we call `.findIndex` on it?

File: src/reducers/exchangeGasFeeReducer.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        (feeInfo) => feeInfo.provider === provider && feeInfo.chain === chain && feeInfo.gasFeeAsset === gasFeeAsset,
+      );
+      if (index !== -1) state.data.splice(index, 1);
```
Comment (IAmKio) [inline src/reducers/exchangeGasFeeReducer.tsx:57]: Is `state.data` supposed to be optional here?

File: src/screens/Bridge/Exchange-CrossChain/Exchange.tsx [hunk 3]
```diff
@@ -101,6 +101,9 @@ function Exchange({ fetchExchangeTitle }: Props) {
   const fromInputRef: any = React.useRef();
   const screenName = getActiveScreenName(navigation);
   const colors = useThemeColors();
+  const gasFeeList = useExchangeGasFee();
+
+  const isEstimaing = gasFeeList?.find((feeInfo) => feeInfo.isEstimating);
```
Comment (IAmKio) [inline src/screens/Bridge/Exchange-CrossChain/Exchange.tsx:106]: As above, can we call `.find` on a potentially `null`? At least it should be `[]`

File: src/screens/Bridge/Exchange-CrossChain/Exchange.tsx [hunk 4]
```diff
@@ -114,6 +117,7 @@ function Exchange({ fetchExchangeTitle }: Props) {
   const [renderItem, setRenderItem] = React.useState(null);
 
   const [selectedProvider, setSelectedProvider] = React.useState('');
+  const [pressToselectedProvider, setPressToSelectedProvider] = React.useState('');
```
Comment (IAmKio) [inline src/screens/Bridge/Exchange-CrossChain/Exchange.tsx:120]:
```suggestion
  const [pressToSelectedProvider, setPressToSelectedProvider] = React.useState('');
```

**Review by IAmKio — CHANGES_REQUESTED** [review]
Just a few small points but other than that - looking good!

**Review by IAmKio — CHANGES_REQUESTED** [review]
Just a few small points but other than that - looking good!

File: src/screens/Bridge/Exchange-CrossChain/Exchange.tsx (hunk 3 above)
Comment (lalitkumarrock) [reply src/screens/Bridge/Exchange-CrossChain/Exchange.tsx:106]: it's []

File: src/screens/Bridge/Exchange-CrossChain/Exchange.tsx (hunk 4 above)
Comment (lalitkumarrock) [reply src/screens/Bridge/Exchange-CrossChain/Exchange.tsx]: Updated it!

**Review by IAmKio — APPROVED** [review]

**Pull request merged**